---
title: "Tutorials"
permalink: /tutorials/tutorial2
layout: default
---
<br>
<br>
<div class="container px-1">

<div class="row">
<div class="col-sm-12 px-3">
<h1><b>Tutorial 2:</b> Characterizing the Rett Transcriptome</h1>
<br>
<h3>Introduction</h3>
<p>In this tutorial, we will analyze an RNA sequencing (RNA-seq) dataset generated by Aldinger <i>et al.</i> from postmortem brain tissue of four female Rett syndrome patients and age-matched female controls 
(<a href = "https://doi.org/10.1038/s41597-020-0527-2" target = "_blank">https://doi.org/10.1038/s41597-020-0527-2</a>). This dataset is already integrated as an example dataset in the ArrayAnalysis app.
Rett syndrome is a rare neurodevelopmental disorder caused by mutations in the X-linked <i>MECP2</i> gene.</p> 
</div>
</div>

<br>
<hr>
<br>

<div class="row">
<div class="col-sm-12 px-3">
<h3>Data upload</h3>
<p>Start the analysis of the data by clicking "Start Analysis" after selecting "RNA-Seq Analysis" and "Raw count". 
In the next tab, click on "Run example".</p>
<br>        
<p>After uploading the data, the metadata and a preview of the count table are shown. In the metadata, you can see that there are four experimental groups: 
the cingulate and temporal cortex of controls and the cingulate and temporal cortex of Rett syndrome patients. Each experimental group includes 4 samples.</p>
</div>
</div>

<div class="row">
<div class="col-sm-12 px-3">
<div class="jumbotron p-5 h-100 bg-white border">
<h4><b>Question 1:</b> How old is the youngest and oldest Rett syndrome (RTT) patient in the dataset?</h4>
<details>
<summary>Show answer</summary>
<br>
<p>16 and 31, respectively. You can find this answer by looking in the metadata tab at the data upload page. You can sort the metadata by age.</p>
</details>
</div>
</div>
</div>

<br>
<hr>
<br>

<div class="row">
<div class="col-sm-12 px-3">
<h3>Pre-processing</h3>
<p>Proceed to the pre-processing step, by clicking "Next" (bottom left of the page) or by clicking the "pre-processing" tab at the navigation bar. 
In this step, you can remove samples, remove lowly expressed genes, and perform normalization. 
Select "Genotype" and "Tissue" as experimental group and start the data pre-processing by clicking on "Calculate".</p>
<details>
<summary>Click here for help</summary>
<br>
 <img 
        src="{{ '/assets/tutorials/tutorial2/experimental_groups.png' | relative_url }}"
        alt = "Experimental groups"
        style="width: 50%; border: none;" 
        title="Experimental groups">
</details>
</div>
</div>

<br>

<div class="row">
<div class="col-sm-12 px-3">
<div class="jumbotron p-5 h-100 bg-white border">
<h4><b>Question 2:</b> Why is normalization needed? How would the boxplots and density plots look like if you would not do any normalization?</h4>
<details>
<summary>Show answer</summary>
<br>
<p>Normalization is needed to adjust for sample-to-sample variability and make the data comparable between samples.
Without normalization the boxplots and density plots would show notable differences in the distribution between samples (see figure below).</p>
<br>
 <img 
        src="{{ '/assets/tutorials/tutorial2/normalization.png' | relative_url }}"
        alt = "Normalization"
        style="width: 70%; border: none;" 
        title="Normalization">
</details>
</div>
</div>
</div>

<br>

<div class="row">
<div class="col-sm-12 px-3">
<div class="jumbotron p-5 h-100 bg-white border">
<h4><b>Question 3:</b> Why is filtering needed? What do we filter for? How many genes passed the applied gene filtering?</h4>
<details>
<summary>Show answer</summary>
<br>
<p>Filtering is needed to remove lowly expressed genes, where the measurements are subject to noise and which cannot be analyzed confidently. 
So, we filter for minimal expression levels (of 10 reads in at least a substantial subset of the samples).
If you apply a threshold of 10, 18,311 genes passed the filtering (see bottom of normalized count table).</p>
<br>
 <img 
        src="{{ '/assets/tutorials/tutorial2/ngenes_qc.png' | relative_url }}"
        alt = "QC-passed genes"
        style="width: 70%; border: none;" 
        title="QC-passed genes">
</details>
</div>
</div>
</div>

<br>

<div class="row">
<div class="col-sm-12 px-3">
<div class="jumbotron p-5 h-100 bg-white border">
<h4><b>Question 4:</b> Look at the correlation plot and PCA plot. Do the samples cluster as expected?</h4> 
<details>
<summary>Show answer</summary>
<br>
<p>In the PCA plot and correlation plot, you can see that there seems to be some clustering by genotype (which would be expected), 
but the clustering is not perfect (two of the clear subclusters consist of one genotype only, one is mixed). 
Moreover, there is almost perfect clustering per patient (which may also be expected), 
but again not perfect (some samples of the same individual donâ€™t cluster together for the two tissues studied).
</p>
</details>
</div>
</div>
</div>


<br>
<hr>
<br>

<div class="row">
<div class="col-sm-12 px-3">
<h3>Differential gene expression analysis</h3>
<p>After we have pre-processed the data, we want to compare the expression profiles of Rett syndrome patients with the controls in the temporal cortex. 
Furthermore, we would like to correct for age and add Ensembl IDs to the output. 
Please select the correct options on the statistical analysis page and click "Calculate".</p>
<details>
<summary>Click here for help</summary>
<br>
 <img 
        src="{{ '/assets/tutorials/tutorial2/settings.png' | relative_url }}"
        alt = "Settings"
        style="width: 30%; border: none;" 
        title="Settings">
</details>
</div>
</div>

<br>

<div class="row">
<div class="col-sm-12 px-3">
<div class="jumbotron p-5 h-100 bg-white border">
<h4><b>Question 5:</b> What is the difference between the p-value and adj. p-value? Which one should you use to find differentially expressed genes? </h4>
<details>
<summary>Show answer</summary>
<br>
<p>The adj. p-value is adjusted for multiple testing. If you test many genes, you will end up with many false positives if you do not correct the p-value. 
For instance, with a 5% significance level for each test, performing 10 tests raises the chance of at least one false positive to about 40%. 
Hence, the adj. p-value should be used to limit the number of false positives and find truly differentially expressed genes.
</p>
</details>
</div>
</div>
</div>

<br>

<div class="row">
<div class="col-sm-12 px-3">
<div class="jumbotron p-5 h-100 bg-white border">
<h4><b>Question 6:</b> Is MECP2 differentially expressed between RTT and IC?</h4>
<details>
<summary>Show answer</summary>
<br>
<p>You can look up genes using the search bar (top right of top table). Here, you can see that MECP2 is not differentially expressed. </p>
 <img 
        src="{{ '/assets/tutorials/tutorial2/mecp2.png' | relative_url }}"
        alt = "MECP2"
        style="width: 80%; border: none;" 
        title="MECP2">
</details>
</div>
</div>
</div>

<br>

<div class="row">
<div class="col-sm-12 px-3">
<div class="jumbotron p-5 h-100 bg-white border">
<h4><b>Question 7:</b> How many genes are downregulated (adj. p-value &lt; 0.05 and log<sub>2</sub>FC &lt; 1)?</h4>
<details>
<summary>Show answer</summary>
<br>
<p>In the <i>Summary</i> tab you can calculate how many genes are up and downregulated for different p-value and log<sub>2</sub>FC thresholds. 
Here you can see that 110 genes are downregulated.</p>
 <img 
        src="{{ '/assets/tutorials/tutorial2/downregulatedgenes.png' | relative_url }}"
        alt = "Downregulated genes"
        style="width: 80%; border: none;" 
        title="Downregulated genes">
</details>
</div>
</div>
</div>

<br>
<hr>
<br>

<div class="row">
<div class="col-sm-12 px-3">
<h3>Gene set analysis</h3>
<p>You can now perform gene set analysis to identify which biological processes and pathways are altered.</p>
</div>
</div>

<br>

<div class="row">
<div class="col-sm-12 px-3">
<div class="jumbotron p-5 h-100 bg-white border">
<h4><b>Question 4:</b> What processes and pathways are altered in Rett syndrome?</h4>
<details>
<summary>Show answer</summary>
<br>
<p>The exact processes and pathways will depend on which gene set collection is used and whether ORA or GSEA is performed. 
However, with both methods, you will find processes and pathways related to inflammation/immune response and synaptic signaling.</p>
</details>
</div>
</div>
</div>
<br>